test/extended/prometheus/collection_profiles.go

```diff
@@ -0,0 +1,257 @@
+	collectionProfileDefault      = collectionProfileFull
+	collectionProfileMinimal      = "minimal"
+	collectionProfileNone         = ""
+
+	operatorName              = "cluster-monitoring-operator"
+	operatorNamespaceName     = "openshift-monitoring"
+	operatorConfigurationName = "cluster-monitoring-config"
+)
+
+var (
+	oc   = exutil.NewCLI(projectName)
+	tctx = context.Background()
+
+	collectionProfilesSupportedList = []string{
+		collectionProfileFull,
+		collectionProfileMinimal,
+	}
+)
+
+type runner struct {
+	kclient                       kubernetes.Interface
+	mclient                       *pov1client.MonitoringV1Client
+	pclient                       pv1.API
+	originalOperatorConfiguration *v1.ConfigMap
+}
+
+var _ = g.Describe("[sig-instrumentation][OCPFeatureGate:MetricsCollectionProfiles] The collection profiles feature-set", g.Ordered, func() {
+	defer g.GinkgoRecover()
+
+	o.SetDefaultEventuallyTimeout(10 * time.Minute)
+	o.SetDefaultEventuallyPollingInterval(5 * time.Second)
+
+	r := &runner{}
+	g.BeforeAll(func() {
+		if !exutil.IsTechPreviewNoUpgrade(oc) {
+			g.Skip("skipping, this feature is only supported on TechPreviewNoUpgrade clusters")
+		}
+		var err error
+		r.kclient, err = kubernetes.NewForConfig(oc.AdminConfig())
+		if err != nil {
+			g.Fail(fmt.Sprintf("failed to create kubernetes client: %v", err))
+		}
+		r.mclient, err = pov1client.NewForConfig(oc.AdminConfig())
+		if err != nil {
+			g.Fail(fmt.Sprintf("failed to create monitoring client: %v", err))
+		}
+		r.pclient = oc.NewPrometheusClient(tctx)
+		operatorConfiguration, err := r.kclient.CoreV1().ConfigMaps(operatorNamespaceName).Get(tctx, operatorConfigurationName, metav1.GetOptions{})
+		o.Expect(err).To(o.BeNil())
+		r.originalOperatorConfiguration = operatorConfiguration
+	})
+
+	g.AfterAll(func() {
+		currentConfiguration, err := r.kclient.CoreV1().ConfigMaps(operatorNamespaceName).Get(tctx, operatorConfigurationName, metav1.GetOptions{})
+		o.Expect(err).To(o.BeNil())
+		currentConfiguration.Data = r.originalOperatorConfiguration.Data
+		_, err = r.kclient.CoreV1().ConfigMaps(operatorNamespaceName).Update(tctx, currentConfiguration, metav1.UpdateOptions{})
+		o.Expect(err).To(o.BeNil())
+	})
+
+	g.It("should apply default collection profile initially", func() {
+		o.Eventually(func() error {
+			monitors, err := r.fetchMonitorsFor([2]string{collectionProfileFeatureLabel, collectionProfileDefault})
+			if err != nil {
+				return err
+			}
+			if len(monitors.Items) == 0 {
+				return fmt.Errorf("no monitors found with collection profile %q", collectionProfileDefault)
+			}
+
+			return nil
+		}).Should(o.BeNil())
+	})
+	g.It("should expose information about the applied collection profile using meta-metrics", func() {
+		vectorExpression := "profile:cluster_monitoring_operator_collection_profile:max{profile=\"%s\"} == 1"
+
+		for _, profile := range collectionProfilesSupportedList {
+			err := r.makeCollectionProfileConfigurationFor(tctx, profile)
```

Comment:
because reconciling the stack takes time on a collection profile change, should we rather change the profile and test all the expectations for this profile in one go?

Reply:
Some cases depend on ordered profile transitions, IIUC this may interfere with changing the design from the current one:
```
<caseA>
	<profileA/B>
	<profileB/A>
<caseB>
	<profileA>
	<profileB/A>
```
to the proposed one,
```
<profileA>
	<caseA>
	<caseB>
<profileB>
	<caseA>
	<caseB>
```
LMK if I missed something.

Reply:
(cc @simonpasquier for a clarification here)

Reply:
I'm no ginko expert, but wouldn't we best represent the profile transistions as ginko containers? So we end up with a structure similar to this:
```
Describe CP tests
  BeforeAll setup make sure default profile is set
  Describe subjects for default profile
    ...
  Describe transistion to minimal
    BeforeEach check or setup minimal
      ...
```

Reply:
I've made two changes to the latest patch (although I still need to take a look at the CI runs to make sure the benchmark results are as expected). These are:

- Based on Jan's review above, encapsulate all cases that start under the default profile in their own context, and same for the minimal profile. Also, ensure this by using `ginkgo.Context`s.
- Drop the bottleneck approach for the 4th test-case (marking a series for deletion, cleaning the tombstones for it, and then polling for change) in favor of checking if the difference between `scrape_samples_scraped` and `scrape_samples_post_metric_relabeling` equal to the number of relabelings mentioned in the `ServiceMonitor` for a particular profile.